### DISCUSSION TITLE
Does React Hook Form reduce rerenders when used with Conrolled Components like MUI TextField

### DISCUSSION THREAD

**Discussion by dakshsaraf** [react-hook-form/react-hook-form#2483] [General] [answered]
![](https://i.stack.imgur.com/fB2pJ.png)

As seen in the above picture react hooks form greatly reduces the rerenders and performs much better than formik. But when used with controlled components from UI libraries like material UI, would it still reduce rerenders and or rerender on every input change like Formik?

**Comment by bluebill1049** [comment, marked as answer]
Yes! totally!

https://react-hook-form.com/api#Controller

https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by dakshsaraf** [reply]
The render count is not changing even after a particular input selection is done. Is there some error in the sandbox

**Reply by bluebill1049** [reply]
nope, because there is no re-render at the form level :)

**Reply by bluebill1049** [reply]
if you don't believe, turn on the dev tool.

**Reply by dakshsaraf** [reply]
To better understand, I am trying to replicate your demo as shown on the website. What am I doing wrong?
Formik: https://codesandbox.io/s/formik-renders-pu528?file=/src/App.js
Reat-Hook-Form: https://codesandbox.io/s/react-hook-form-renders-umzuv?file=/src/App.js

**Reply by bluebill1049** [reply]
your example looks correctly with RHF :)

**Reply by dakshsaraf** [reply]
There are two bugs to my examples.
Formik: The count is not increasing
React-Hook-Form: if the email field is not validated rerenders happen at every inputchange for the username field, why is that the case?

**Reply by bluebill1049** [reply]
formik 
```
        count++;
        console.log("rendering.." + count);
```

hook form: i am not sure why you get re-render, it should only trigger correct the input.

**Reply by bluebill1049** [reply]
<img width="1024" alt="Screen Shot 2020-08-05 at 7 09 10 pm" src="https://user-images.githubusercontent.com/10513364/89394220-26b4e900-d74f-11ea-9c77-7bbcd0208773.png">

**Reply by dakshsaraf** [reply]
Yes I refreshed it and it started working, do not know why that was happening before. Thanks so much for taking out so much time to help me

**Reply by dakshsaraf** [reply]
I am shifting my project from formik to react-hook-form. I had a multistep form in formik with over 100+ fields that was too slow despite using FastField. I am certain react-hook-form would solve the performance issue. Great Library
